tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts.py

```diff
@@ -284,6 +285,48 @@ def test_add_new_file():
     }
 
 
+def test_add_changed():
+    artifact = wandb.Artifact(type="dataset", name="my-arty")
+    with artifact.new_file("file1.txt") as f:
+        f.write("hello")
+
+    entry1 = artifact.manifest.entries["file1.txt"]
+    assert entry1.path == "file1.txt"
+    assert Path(entry1.local_path).read_text() == "hello"
+
+    with artifact.new_file("file1.txt") as f:
+        f.write("goodbye")
+
+    entry2 = artifact.manifest.entries["file1.txt"]
+    assert entry1 != entry2
+    assert entry2.path == "file1.txt"
+    assert entry1.local_path != entry2.local_path
+    assert Path(entry2.local_path).read_text() == "goodbye"
+
+
+def test_add_changed_by_name(monkeypatch, tmp_path):
+    monkeypatch.setenv("WANDB_DATA_DIR", str(tmp_path))
```

Comment:
This won't have any effect if another test already called `get_staging_dir`. Maybe just mock `get_staging_dir`.